roles/postgres/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+- name: Install postgres Python library.
+  apt:
+    name: "python3-psycopg2"
+    state: present
+
+- name: Install postgres packages.
+  apt:
+    name:
+      - postgresql
+      - postgresql-contrib
+      - libpq-dev
+    state: present
+
+- name: Ensure PostgreSQL data directory exists.
+  file:
+    path: "{{ postgresql_data_dir }}"
+    owner: "{{ postgresql_user }}"
+    group: "{{ postgresql_group }}"
+    state: directory
+    mode: 0700
+
+- name: Check if postgres is initialised.
+  stat:
+    path: "{{ postgresql_data_dir }}/PG_VERSION"
+  register: pgdata_dir_version
+
+- name: Initialise postgres if not already.
+  command: "{{ postgresql_bin_path }}/initdb -D {{ postgresql_data_dir }}"
+  when: not pgdata_dir_version.stat.exists
+  become: true
+  become_user: "{{ postgresql_user }}"
+  # See: https://github.com/ansible/ansible/issues/16048#issuecomment-229012509
+  vars:
+    ansible_ssh_pipelining: true
+
+- name: Check postgres is started and enabled on boot.
+  service:
+    name: "{{ postgresql_daemon }}"
+    state: "{{ postgresql_service_state }}"
+    enabled: "{{ postgresql_service_enabled }}"
+
+- name: Add postgres users.
+  community.postgresql.postgresql_user:
+    name: "{{ item.name }}"
+    login_user: "{{ item.login_user }}"
+    role_attr_flags: "{{ role_attr_flags }}"
+  with_items: "{{ postgresql_users }}"
+  become: true
+  become_user: "{{ postgresql_user }}"
+  vars:
+    ansible_ssh_pipelining: true
+
+- name: Add postgres databases.
+  community.postgresql.postgresql_db:
+    name: "{{ item.name }}"
+    owner: "{{ item.owner | default(postgresql_user) }}"
+  with_items: "{{ postgresql_databases }}"
+  become: true
+  become_user: "{{ postgresql_user }}"
+  vars:
+    ansible_ssh_pipelining: true
+
+- name: Check postgres is running.
+  command: "{{ postgresql_bin_path }}/pg_ctl -D {{ postgresql_data_dir }} status"
+  changed_when: false
+  become: true
+  become_user: "{{ postgresql_user }}"
```

Comment:
What's the use of this?

Reply:
Afaik sudo doesn't have a role on psql by default, so this runs it under the `postgres` user

Reply:
I highlighted multiple lines not just the last one. I'm asking why this entire step is necessary.

Reply:
It's mostly useful on subsequent runs of the role, when all of the above is skipped due to already existing, this ensures psql is still running